File: Objective-C/Reference App/5. Basic Map with TGMapView (no Storyboard)/InsertMapViewWithoutStoryboardViewController.m
//
//  InsertMapViewWithoutStoryboardViewController.m
//  Reference App
//
//  Created by David Deller on 5/3/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

#import "InsertMapViewWithoutStoryboardViewController.h"

@interface InsertMapViewWithoutStoryboardViewController ()

@end

@implementation InsertMapViewWithoutStoryboardViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Part 1: Initialize the view
    TGMapView *mapView = [TGMapView.alloc initWithFrame:self.view.bounds];
    mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    mapView.translatesAutoresizingMaskIntoConstraints = YES;
    
    // Part 2: Customize it to your liking
    mapView.tintColor = UIColor.greenColor;
    mapView.zoomLevel = 12;
    mapView.showsUserLocation = YES;
    [mapView setUserTrackingMode:MGLUserTrackingModeFollow animated:NO];
    
    // Part 3: Add it to your existing view
    [self.view addSubview:mapView];
}

@end

File: Swift 4/Reference App/10. Insert Map View (without Storyboard)/InsertMapViewWithoutStoryboardViewController.swift
//
//  InsertMapViewWithoutStoryboardViewController.swift
//  Reference App
//
//  Created by David Deller on 5/1/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

import UIKit
import TallyGoKit

class InsertMapViewWithoutStoryboardViewController: ExampleViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Part 1: Initialize the view
        let mapView = TGMapView(frame: view.bounds)
        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        mapView.translatesAutoresizingMaskIntoConstraints = true
        
        // Part 2: Customize it to your liking
        mapView.tintColor = UIColor.green
        mapView.zoomLevel = 12
        mapView.showsUserLocation = true
        mapView.setUserTrackingMode(.follow, animated: false)
        
        // Part 3: Add it to your existing view
        view.addSubview(mapView)
    }

}
